Java/syntax_test_java.java

```diff
@@ -312,3 +371,502 @@ void test2() {
 //            ^ support.class
     }
 }
+
+public enum FooEnum {
+//     ^^^^ storage.type.java
+  FOO;
+//^^^ constant.other.enum
+}
+// <- meta.class.java meta.class.body.java punctuation.section.class.end.java
```

Comment:
Generally we don't do things like `punctuation.section.class`, but instead `punctuation.section.block`. Then the color scheme author can target class braces via `meta.class punctuation.section.block`.